File: backend/app/schemas/log.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class LogRecord(BaseModel):
    """日志记录"""
    id: int
    user_id: Optional[int] = None
    nickname: Optional[str] = None
    ip_address: Optional[str] = None
    login_location: Optional[str] = None
    os: Optional[str] = None
    browser: Optional[str] = None
    login_time: datetime
    created_at: datetime

    model_config = {"from_attributes": True}


class LogList(BaseModel):
    """日志列表"""
    list: list[LogRecord]
    total: int
    page: int
    page_size: int


class OperationLogRecord(BaseModel):
    """操作日志记录"""
    id: int
    nickname: Optional[str] = None
    ip_address: Optional[str] = None
    login_location: Optional[str] = None
    os: Optional[str] = None
    browser: Optional[str] = None
    login_time: datetime
    operation: Optional[str] = None
    operation_type: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/log.ts
import { http } from '@/utils/request'
import type { PaginationParams, PaginationData } from '@/types/common'

export interface LogRecord {
  id: number
  userNickname: string
  ip: string
  location: string
  os: string
  browser: string
  loginTime: string
  operation: string
}

export const logApi = {
  getList(params: PaginationParams & { 
    dateRange?: [string, string]
    userId?: number
    ip?: string
    operationType?: string
  }) {
    // 转换参数名以匹配后端 API
    const apiParams: any = {
      page: params.page,
      page_size: params.pageSize,
    }
    if (params.dateRange) {
      apiParams.start_date = params.dateRange[0]
      apiParams.end_date = params.dateRange[1]
    }
    if (params.userId) apiParams.user_id = params.userId
    if (params.ip) apiParams.ip_address = params.ip
    if (params.operationType) apiParams.operation_type = params.operationType
    
    return http.get<PaginationData<LogRecord>>('/v1/log/login', { params: apiParams })
  }
}
